initial-promises/node-server/inventory/1.0/test-inventory.pl

```diff
@@ -0,0 +1,95 @@
+#!/usr/bin/perl
+
+use warnings;
+use strict;
+use XML::TreePP;
+use Data::Dumper;
+
+my $file = $ARGV[0];
+if(!defined $file || $file eq "") {
+  print "Usage check-inventory <filename>\n";
+  exit 2;
+}
+
+my $tpp = XML::TreePP->new();
+my $tree = $tpp->parsefile( $file );
+my $os = $tree->{REQUEST}->{CONTENT}->{OPERATINGSYSTEM};
+
+# Test extension
+my $rudder = $tree->{REQUEST}->{CONTENT}->{RUDDER};
+if(!defined $rudder) {
+  print "ERROR: No RUDDER extension\n";
+  exit 1;
+}
+
+# Test RUDDER/UUID
+my $uuid = $rudder->{UUID};
+if(!defined $uuid || $uuid eq "" || ($uuid ne "root" && $uuid !~ /[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}/i)) {
+  print "ERROR: No valid UUID\n";
+  exit 1;
+}
+
+# Test RUDDER/HOSTNAME or OPERATINGSYSTEM/FQDN
+my $hostname = $rudder->{HOSTNAME};
+my $fqdn = $os->{FQDN};
+if( (!defined $hostname || $hostname eq "") && (!defined $fqdn || $fqdn eq "")) {
+  print "ERROR: No RUDDER/HOSTNAME or OPERATINGSYSTEM/FQDN\n";
+  exit 1;
+}
+
+# more than one agent are possible
+my @agents;
+if(!defined $rudder->{AGENT}) {
+  print "ERROR: No RUDDER/AGENT\n";
+  exit 1;
+} elsif(ref($rudder->{AGENT}) eq 'HASH') {
+  # One RUDDER/AGENT
+  @agents = ($rudder->{AGENT});
+} else {
+  # Many RUDDER/AGENT
+  @agents = @{$rudder->{AGENT}};
+}
+foreach my $agent (@agents) {
+  # Test RUDDER/AGENT/OWNER
+  my $owner = $agent->{OWNER};
+  if(!defined $owner || $owner eq "") {
+    print "ERROR: No RUDDER/AGENT/OWNER\n";
+    exit 1;
+  }
+  
+  # Test RUDDER/AGENT/POLICY_SERVER_UUID
+  my $server_uuid = $agent->{POLICY_SERVER_UUID};
+  if(!defined $server_uuid || $server_uuid eq "" || ($server_uuid ne "root" && $server_uuid !~ /[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}/i)) {
+    print "ERROR: No RUDDER/AGENT/POLICY_SERVER_UUID\n";
+    exit 1;
+  }
+  
+  # Test RUDDER/AGENT/AGENT_NAME
+  my $agent_name = $agent->{AGENT_NAME};
+  if(!defined $agent_name || $agent_name eq "") {
+    print "ERROR: No RUDDER/AGENT/AGENT_NAME\n";
+    exit 1;
+  }
+}
+
+# Test OPERATINGSYSTEM/{FULL_NAME or KERNEL_NAME or NAME}
+if(!defined $os->{FULL_NAME} || !defined $os->{KERNEL_NAME} || !defined $os->{NAME}) {
+  print "ERROR: No OPERATINGSYSTEM/FULL_NAME nor OPERATINGSYSTEM/KERNEL_NAME nor OPERATINGSYSTEM/NAME\n";
+  exit 1;
+}
+
+# Test OPERATINGSYSTEM/KERNEL_VERSION or if OPERATINGSYSTEM/KERNEL_NAME.toLowerCase == aix: HARDWARE/OSVERSION
+if(defined $os->{KERNEL_NAME} && lc($os->{KERNEL_NAME}) eq "aix") {
+  if(!defined $tree->{REQUEST}->{CONTENT}->{HARDWARE}->{OSVERSION}) {
+    print "ERROR: No HARDWARE/OSVERSION\n";
+    exit 1;
+  }
+} else {
+  if(!defined $os->{KERNEL_VERSION}) {
+    print "ERROR: No OPERATINGSYSTEM/KERNEL_VERSION\n";
```

Comment:
error messages could be a bit more helpful, like 
ERROR: No valid OPERATINGSYSTEM/KERNEL_VERSION entry in the inventory